Clarify custom MeterTag value expression resolver docs (#7480)
micrometer-metrics/micrometer: docs/modules/ROOT/pages/concepts/counters.adoc

include::{include-java}/metrics/CountedAspectTest.java[tags=meter_tag_annotation_handler,indent=0]
-----

> NOTE: `CustomValueExpressionResolver` is an application-provided `ValueExpressionResolver` example. Micrometer does not provide this class. You need to provide an implementation that matches your application framework if you use `@MeterTag` expressions.

Reviewer: I would reword this a bit. While Micrometer doesn't provide an implementation, the end-user doesn't _necessarily_ need to provide one. The framework they're using or some other third-party library may provide one.

```suggestion
NOTE: `CustomValueExpressionResolver` is a sample implementation of `ValueExpressionResolver` for documentation purposes. Micrometer does not provide this class. An implementation is required to use `@MeterTag` expressions.
```

Author: Updated the note so it allows resolvers provided by frameworks or third-party libraries in a757bac.
